docker/docker.mk

```diff
@@ -8,6 +8,9 @@
 DOCKER:=docker
 DOCKERFILE_DIR?=./docker
 
+DAPR_SYSTEM_IMAGE_NAME=$(RELEASE_NAME)
```

Comment:
Does one image will contain everything including daprd and the other one will contain only daprd?
Is it a long term goal to put each system service into a separate image?

Reply:
oh I missed your comment. Yes, the current image(dapr) will contain all binaries for backward compat as is. 

In a long-term goal, yes we will separate it completely.

Reply:
We might separate it in the future, but I don't see a reason to do that unless we grow > 1GB or something similar.

The data plane (sidecar) is important as it gets scaled in and out and should be able to pull the images quickly.
This is not true for the control plane which remains relatively static, and thus has (currently) no real reason for separation.

Reply:
yes, I 100% agree with the reasoning for taking out daprd.

Here is what my reasoning is to separate others out as well:
1. A bug in operator is fixed and the docker image is pushed.
2. But images for sentry, placement will also be pulled down(whenever they move to a new node) and updated where as its not needed, this can be completely avoided if all are separated.

Reply:
Well, if only the operator is fixed, only the Docker layer containing the operator will be pulled down. A fix for the operator will only cause the layer containing the operator to download. Even in the case that all layers download again, non-Operator control plane items will be unchanged and not contain any modified code.

Reply:
hmm If i am not misunderstanding `docker image` layer, operator and the other system components are in the same layer. 

If user uses helm chart to upgrade, we can handle via helm chart version...

Reply:
We can separate them to different layers, but we really don't have to. Like I explained, the image size is so small now that re-pulling everything is not an issue, and a hotfix in one of the control plane components will not cause code modifications in the others.

Reply:
ok, so an update to operator will result in a new image to be pushed and it would mean new image download by docker for placement, sentry which an be avoided. Given the size is small and its an optimization, I am fine postponing it for now, lets not close the issue and keep it open to track segregating them.

Reply:
Ok. it will result in re-pulling if placement or sentry pods get restarted.

Reply:
@amanbha I would like to create new issue to create system process image separately. #633 focuses only on creating daprd image.

Reply:
@youngbupark sure, sounds good, as long as we have a issue, its good.

Reply:
> @youngbupark sure, sounds good, as long as we have a issue, its good.

@amanbha I created  https://github.com/dapr/dapr/issues/1344